platform/mds/mds-web/src/main/resources/webapp/js/controllers.js

```diff
@@ -3865,13 +3890,35 @@
                 callback);
         };
 
-        $scope.setAvailableFieldsForDisplay = function() {
+        $scope.subclassCurrent = null;
+
+        $scope.setAvailableFieldsForDisplay = function(scope) {
             var i;
-            $scope.availableFieldsForDisplay = [];
-            for (i = 0; i < $scope.allEntityFields.length; i += 1) {
-                if (!$scope.allEntityFields[i].nonDisplayable) {
-                    $scope.availableFieldsForDisplay.push($scope.allEntityFields[i]);
+            var availableFieldsForDisplay = [];
+            for (i = 0; i < scope.allEntityFields.length; i += 1) {
+                if (!scope.allEntityFields[i].nonDisplayable) {
+                    availableFieldsForDisplay.push(scope.allEntityFields[i]);
+                }
+            }
+            var subclassnameCurrent = scope.$parent.$parent.$parent.subclassCurrent;
+            if(subclassnameCurrent) {
+                var entitiesT = scope.$parent.$parent.$parent.entitiesDerived;
```

Comment:
suffix T: "Temporary"
suffix I: "Internal"

Both are a symbol whose main significance is in avoiding a naming conflict.  Both should have a convention.  Above is a convention I have used before.  Please let me know if you have a preferred choice.